### ISSUE TITLE
read_routes - catch ioctl error for loopback device

### ISSUE THREAD

**Issue by speakinghedge** [secdev/scapy#957]
if the looback device has no address assigned the [ioctl](https://github.com/secdev/scapy/blob/master/scapy/arch/linux.py#L227) used for fetching the interface address fails and scapy exits.

also see #938

**Comment by speakinghedge** [comment]
I will take care for this. As this kind of error is common for a network namespaced environments I would like to add a warning that guides the user, e.g. 

```
warning: ioctl on loopback device failed - are you running this in a network namespace and the device is down?
```

any objections?

**Comment by guedou** [comment]
Thanks for your help !

I think that the message will be better if you remove the part related to namespace, and the loopback. Specifying that the device is likely down is more than enough.

Could you also add a unit test in test/linux.uts ? The last test is a good start to implement yours.

**Comment by speakinghedge** [comment]
> Specifying that the device is likely down is more than enough.

The state of the interface is not the problem. It could be up and the user removed the address manually (in the context of the loopback a bad idea - but still possible).

I think the main issue with the current solution is the misleading error message
> IOError: [Errno 99] Cannot assign requested address

I will catch 99 and show 'no address assigned' and dump the plain strerrno message for all other errors.

> Could you also add a unit test in test/linux.uts ? The last test is a good start to implement yours.

Of course. But testing this without removing the address of the lo-device (that could interrupt/bring down local services on the test host) or using a namespace (and moving the test process to it) is not possible. Both solutions are ugly and bloated. I could move the LOOPBACK_NAME to the arguments of read_routes - for the sake of becoming testable (regarding to the current issue).

**Comment by guedou** [comment]
Closed by #958
